helper/forwarding/util.go

```diff
@@ -56,11 +58,31 @@ func GenerateForwardedHTTPRequest(req *http.Request, addr string) (*http.Request
 }
 
 func GenerateForwardedRequest(req *http.Request) (*Request, error) {
+	var reader io.Reader = req.Body
+	ctx := req.Context()
+	maxRequestSize := ctx.Value("max_request_size")
+	if maxRequestSize != nil {
+		max, ok := maxRequestSize.(int64)
+		if !ok {
+			return nil, errors.New("could not parse max_request_size from request context")
+		}
+		if max > 0 {
+			reader = io.LimitReader(req.Body, max)
+		}
+	}
+
+	buf := bytes.NewBuffer(nil)
```

Comment:
Could use` ioutil.ReadAll()` here since you want `[]byte` later.

Reply:
That's fair, I just kept that part of the logic that was there. Feel free to pull down the branch and change it yourself if you like, too!

Reply:
Done!